app/controller/dashboard.js

```diff
@@ -1,75 +1,63 @@
+// Imports for packages
 import Hubkit from 'hubkit';
 import Promise from 'bluebird';
 import _ from 'lodash';
 import ini from 'ini';
 
+// Local Imports
 import app from '~/';
 import { isBeta } from './auth';
 import { Application } from '~/model/application';
 
-// Create Hubkit with a static instance
-let gh = new Hubkit();
+// Global instances
+const gh = new Hubkit();
 
-// wrap for top-level error handling with async functions
-// strongloop.com/strongblog/async-error-handling-expressjs-es7-promises-generators/
-let wrap = fn => (...args) => fn(...args).catch(args[2]);
-
-// returns `true` if the repo has a .apphub file
-// in its top-level directory, otherwise returns `false`
-function repoHasAppHubFile(repo, token) {
-  return gh.request(`GET /repos/${repo}/contents/.apphub`, {token})
-  .then(() => true)
-  .catch(err => {
-    // no .apphub file
-    if (err.status == 404) return false;
-    // some other error
-    throw err;
-  });
-}
-
-app.get('/dashboard', isBeta, wrap(async (req, res, next) => {
-  const repos = await gh.request('GET /user/repos', {
+app.get('/dashboard', isBeta, (req, res, next) => {
+  const token = req.user.github.accessToken;
+  const repos = gh.request('GET /user/repos', {
     type: 'public',
-    token: req.user.github.accessToken,
+    token,
+  })
+  .catch(error => {
+    return res.render('error', {
+      message: 'Houston incountered a problem communicating with Github',
+      stack: error,
+    });
   });
 
-  const appHubFileResults = await Promise.all(
-    repos.map(repo => repoHasAppHubFile(
-      repo.full_name,
-      req.user.github.accessToken
-    ))
-  );
-
-  const applications = await Promise.all(_.chain(_.zip(repos, appHubFileResults))
-    .filter(([repo, hasAppHubFile]) => hasAppHubFile)
-    .map(([repo, hasAppHubFile]) => {
-      return Application.findOne({
-       'github.owner': repo.owner.login,
-       'github.name': repo.name,
-      })
-      .then(appFromDB => appFromDB || Promise.resolve(new Application({
+  Promise.filter(repos, repo => {
+    return gh.request(`GET /repos/${repo.full_name}/contents/.apphub`, {token})
+    .then(() => true)
+    .catch(() => false);
+  })
+  .map(repo => {
```

Comment:
really nice refactoring here, it's a lot more conceptually simple (rightfully so) than what I came up with (especially for people who aren't familiar with functional programming idioms).